indico/modules/events/registration/util.py

```diff
@@ -217,29 +218,34 @@ def check_registration_email(regform, email, registration=None, management=False
             return dict(status='ok', user=None)
 
 
-def make_registration_form(regform, management=False, registration=None):
-    """Create a WTForm based on registration form fields."""
+def make_registration_schema(regform, management=False, registration=None):
+    """Dynamically create a Marshmallow schema based on the registration form fields."""
+    class RegistrationSchema(IndicoSchema):
+        class Meta:
+            unknown = EXCLUDE
 
-    class RegistrationFormWTF(IndicoForm):
-        if management:
-            notify_user = BooleanField(_('Send email'), widget=SwitchWidget())
-
-        def validate_email(self, field):
-            status = check_registration_email(regform, field.data, registration, management=management)
+        @validates('email')
+        def validate_email(self, email, **kwargs):
+            status = check_registration_email(regform, email, registration, management=management)
             if status['status'] == 'error':
-                raise ValidationError('Email validation failed: ' + status['conflict'])
+                raise MMValidationError('Email validation failed: ' + status['conflict'])
+
+    schema = {}
+
+    if management:
+        schema['notify_user'] = fields.Boolean()
 
     for form_item in regform.active_fields:
         if not management and form_item.parent.is_manager_only:
             continue
 
         field_impl = form_item.field_impl
-        setattr(RegistrationFormWTF, form_item.html_field_name, field_impl.create_wtf_field())
-    signals.event.registration_form_wtform_created.send(regform, registration=registration, management=management,
-                                                        wtform_cls=RegistrationFormWTF)
+        # Some field validators require access to the previous registration
+        schema[form_item.html_field_name] = field_impl.create_mm_field(registration=registration)
+
+    # TODO: what to do with signal -> signals.event.registration_form_wtform_created
```

Comment:
Check indico/indico#4642 for the rationale why it was added - so I think in the end we should probably provide a similar signal that has access to the schema in order to modify that one if needed. But I think we could consider getting rid of it; if the UN folks (cc @omegak) still need it when adapting to the react regform (I'd still like to provide some way for plugins to register new regform field types!) we can see what's the best way of restoring it (or let them send another PR)

Reply:
We are, indeed, still needing this signal (or something equivalent). You can see here how we are using it:
https://github.com/UNOG-Indico/indico-un/blob/master/un/modules/registration/forms.py#L60-L61